File: tests/fixtures/vulnerable/jwt_hardcoded.py
"""Vulnerable: hardcoded JWT secrets."""
import os
import jwt

# Hardcoded secret assignment
JWT_SECRET = "supersecretkey123"
SECRET_KEY = "changeme"
SIGNING_KEY = "my-jwt-signing-key"

# Fallback default in os.environ.get
TOKEN_SECRET = os.environ.get("TOKEN_SECRET", "fallback-secret-value")

# JWT encode with literal secret
def create_token(user_id):
    return jwt.encode({"user_id": user_id}, "hardcoded-secret", algorithm="HS256")

File: tests/fixtures/vulnerable/jwt_hardcoded.ts
// Vulnerable: hardcoded JWT secrets in TypeScript
import jwt from "jsonwebtoken";

const JWT_SECRET = "super-secret-key-123";
const SIGNING_KEY = "my-signing-key";

// Fallback default
const TOKEN_SECRET = process.env.TOKEN_SECRET || "fallback-secret";

export function createToken(userId: string): string {
  return jwt.sign({ userId }, "hardcoded-inline-secret", { expiresIn: "1h" });
}
